libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -168,10 +175,25 @@ bool AP_Arming::barometer_checks(bool report)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_BARO)) {
-        if (!AP::baro().all_healthy()) {
+        const AP_Baro &baro = AP::baro();
+        if (!baro.all_healthy()) {
             check_failed(ARMING_CHECK_BARO, report, "Barometer not healthy");
             return false;
         }
+        if (_baro2gps == 1 && checks_to_perform & ARMING_CHECK_GPS) {
```

Comment:
You need to also check ARMING_CHECK_ALL, see the if for baro a couple lines above.

Nitpicking: no need to check the value is 1, as long it isn't zero, it should go in.